packages/core/sentry.gradle

```diff
@@ -4,7 +4,8 @@ import java.util.regex.Matcher
 import java.util.regex.Pattern
 
 project.ext.shouldSentryAutoUpload = { ->
-  return System.getenv('SENTRY_DISABLE_AUTO_UPLOAD') != 'true'
+  return System.getenv('SENTRY_DISABLE_AUTO_UPLOAD') != 'true' ||
+         System.getenv('SENTRY_DISABLE_NATIVE_DEBUG_UPLOAD') != 'true'
```

Comment:
Hey @lucas-zimerman 👋 
While checking some [a CI checks issue on a dependandabot PR](https://github.com/getsentry/sentry-react-native/pull/4258#issuecomment-2468504313) I revisited the Android logic here. I think for the new flag to work as on iOS we should change `||` with `&&`. Wdyt?

```
project.ext.shouldSentryAutoUpload = { ->
  return System.getenv('SENTRY_DISABLE_AUTO_UPLOAD') != 'true' &&
         System.getenv('SENTRY_DISABLE_NATIVE_DEBUG_UPLOAD') != 'true'
}
```

or 
```
project.ext.shouldSentryAutoUpload = { ->
  return !(System.getenv('SENTRY_DISABLE_AUTO_UPLOAD') == 'true' ||
         System.getenv('SENTRY_DISABLE_NATIVE_DEBUG_UPLOAD') == 'true')
}
```